Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -6433,7 +6437,10 @@ void WebPageProxy::decidePolicyForResponse(IPC::Connection& connection, FrameIde
 {
     RefPtr frame = WebFrameProxy::webFrame(frameID);
     MESSAGE_CHECK_BASE(frame, &connection);
-    decidePolicyForResponseShared(Ref { frame->process() }, internals().webPageID, frameID, WTFMove(frameInfo), identifier, navigationID, response, request, canShowMIMEType, downloadAttribute, listenerID);
+    if (auto provisionalFrameProxy = frame->provisionalFrame())
+        decidePolicyForResponseShared(Ref { provisionalFrameProxy->process() }, internals().webPageID, frameID, WTFMove(frameInfo), identifier, navigationID, response, request, canShowMIMEType, downloadAttribute, listenerID);
```

Comment:
This is a lot of duplicate code.  Could we do something like this instead?
auto* provisionalFrame = frame->provisionalFrame();
Ref sourceProcess { provisionalFrame ? provisionalFrame->process() : frame->process() };
decidePolicyForResponseShared(sourceProcess, ...)